### ISSUE TITLE
Clarification on algorithm

### ISSUE THREAD

**Issue by bkj** [automl/HpBandSter#10]
I'm looking at the `example_4_rnn_20_newsgroups` example, and it looks like you train N models for 9 epochs, then take the top N/2 models and _retrain them from scratch_ for 27 epochs.  Is there an algorithmic reason why you wouldn't want to "hot start" the 27 epoch models, so you'd only have to train them for 18 epochs?  Or is this something we should be doing in general?

~ Ben

**Comment by sfalkner** [comment]
For this particular example, I was just to lazy to implement storing and loading the model. In general, you could warm start the model and continue training. There are some scenarios though, where this does not apply, e.g.:
- when the learning rate schedule depends on the number of epochs
- when you expect rather noisy runs, retraining from scratch gives a more robust measure
- when the budget is not epochs, but dataset subsamples, such that continued training does not make a lot of sense

Best,
Stefan

**Comment by bkj** [comment]
Great -- makes sense. Thanks!
